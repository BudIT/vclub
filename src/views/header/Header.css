.container {
  display: flex;
  justify-content: space-between;
  padding: 0 10px;
  background-color: #353535;
  color: #97A6A6;
  fill: #97A6A6;
}

.title {
  margin-top: 10px;
  color: inherit;;
  font-weight: bold;
}

.panel {
  color: inherit;
  fill: inherit;
}
