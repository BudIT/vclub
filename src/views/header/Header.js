import React, { PropTypes } from 'react';
import compose from 'recompose/compose';
import { connect } from 'react-redux';

// subviews
import HeaderRight from './HeaderRight/HeaderRight';
import HeaderLeft from './HeaderLeft/HeaderLeft';

import style from './Header.css';

const enhance = compose(
  connect(state => ({
    numberOfMembers: state.members.length,
    currentRoomName: state.rooms.currentRoom,
    user: state.auth.user,
  })),
);

function Header(props) {
  const roomsNames = ['SHARING', 'VIDEO', 'CHAT', 'WHITEBOARD'];

  const {
    dispatch,
    numberOfMembers,
    currentRoomName,
    user,
  } = props;


  return (
    <div className={style.header}>
      <HeaderLeft
        currentRoomName={currentRoomName}
        roomsNames={roomsNames}
        user={user}
        dispatch={dispatch}
      />
      <HeaderRight
        numberOfMembers={numberOfMembers}
        dispatch={dispatch}
      />
    </div>
  );
}

Header.propTypes = {
  dispatch: PropTypes.func.isRequired,
  numberOfMembers: PropTypes.number.isRequired,
  currentRoomName: PropTypes.string.isRequired,
  user: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    master: PropTypes.bool.isRequired,
  }).isRequired,
};

export default enhance(Header);
