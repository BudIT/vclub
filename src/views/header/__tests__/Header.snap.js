import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../Header';

const currentRoomName = 'VIDEO';
const numberOfMembers = 5;

test('<Header /> renders correctly', () => {
  const dispatchSpy = jest.fn();
  const rendered = renderer.create(
    <Header
      currentRoomName={currentRoomName}
      numberOfMembers={numberOfMembers}
      dispatch={dispatchSpy}
    />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
