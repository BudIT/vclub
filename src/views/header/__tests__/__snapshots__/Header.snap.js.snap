exports[`test <Header /> renders correctly 1`] = `
<div
  className="header">
  <ul
    className="ul">
    <li>
      <button
        className="tab"
        onClick={[Function]}>
        SHARING
      </button>
    </li>
    <li>
      <button
        className="tabCurrent"
        onClick={[Function]}>
        VIDEO
      </button>
    </li>
    <li>
      <button
        className="tab"
        onClick={[Function]}>
        CHAT
      </button>
    </li>
    <li>
      <button
        className="tab"
        onClick={[Function]}>
        WHITEBOARD
      </button>
    </li>
  </ul>
  <ul
    className="ul">
    <li>
      <button
        className="tab"
        onClick={[Function]}>
        ⛹ 
        5
      </button>
    </li>
    <li
      className="dropdown">
      <button
        className="tab">
        ☰
      </button>
      <div
        className="dropdownContent">
        <button
          className="tab"
          onClick={[Function]}>
          Log Out
        </button>
      </div>
    </li>
  </ul>
</div>
`;
